src/compiler/parser.ts

```diff
@@ -3974,7 +3999,7 @@ namespace ts {
             // a => (b => c)
             // And think that "(b =>" was actually a parenthesized arrow function with a missing
             // close paren.
-            if (!fillSignature(SyntaxKind.ColonToken, isAsync, node) && !allowAmbiguity) {
+            if (!fillSignature(SyntaxKind.ColonToken, isAsync, node, allowAmbiguity) && !allowAmbiguity) {

```

Comment:
I'm looking into rebasing this PR, I got reasonably far, but all of the signature stuff has been completely removed from the codebase - so I'm going to wait on this till our pairing session